examples/blockchain/network.rs

```diff
@@ -0,0 +1,377 @@
+        match recipient {
+            Some(node_ix) => {
+                if let Some(peer_id) = self.peer_by_index.get(&node_ix) {
+                    self.rq_rp.send_request(peer_id, message);
+                } else {
+                    debug!(target: "Blockchain-network", "No peer_id known for node {:?}.", node_ix);
+                }
+            }
+            None => {
+                for peer_id in self.peers.iter() {
+                    self.rq_rp.send_request(peer_id, message.clone());
+                }
+            }
+        }
+    }
+
+    fn send_block_message(&mut self, block: Block) {
+        debug!(target: "Blockchain-network", "Sending block message num {:?}.", block.num);
+        let message = Message::Block(block).encode();
+        for peer_id in self.peers.iter() {
+            self.rq_rp.send_request(peer_id, message.clone());
+        }
+    }
+}
+
+impl NetworkBehaviourEventProcess<MdnsEvent> for Behaviour {
+    fn inject_event(&mut self, event: MdnsEvent) {
+        if let MdnsEvent::Discovered(list) = event {
+            let auth_message = Message::Auth(self.node_ix).encode();
+            for (peer, _) in list {
+                if self.peers.iter().any(|p| *p == peer) {
+                    continue;
+                }
+                self.peers.push(peer);
+                self.rq_rp.send_request(&peer, auth_message.clone());
+            }
+        }
+    }
+}
+
+impl NetworkBehaviourEventProcess<RequestResponseEvent<Request, Response>> for Behaviour {
+    fn inject_event(&mut self, event: RequestResponseEvent<Request, Response>) {
+        if let RequestResponseEvent::Message {
+            peer: peer_id,
+            message,
+        } = event
+        {
+            match message {
+                RequestResponseMessage::Request {
+                    request_id: _,
+                    request,
+                    channel,
+                } => {
+                    let message = Message::decode(&mut &request[..])
+                        .expect("honest network data should decode");
+                    match message {
+                        Message::Consensus(msg) => {
+                            self.consensus_tx.unbounded_send(msg).unwrap();
+                        }
+                        Message::Auth(node_ix) => {
+                            debug!(target: "Blockchain-network", "Authenticated peer: {:?} {:?}", node_ix, peer_id);
+                            self.peer_by_index.insert(node_ix, peer_id);
+                        }
+                        Message::Block(block) => {
+                            debug!(target: "Blockchain-network", "Received block num {:?}", block.num);
+                            self.block_tx.unbounded_send(block).unwrap();
+                        }
+                    }
+                    // Not sure if better to "send" () or just do nothing?
+                    self.rq_rp.send_response(channel, ()).unwrap();
```

Comment:
I would do nothing.

Reply:
Ok, doing nothing seems to work well, so we do nothing.